lib/puppet/environments.rb

```diff
@@ -40,6 +40,10 @@ def get!(name)
     end
 
     def clear_all
+      # This method was implemented by future parser, but doesn't ever seem
+      # to be called, but might? be called via manual eviction.
+      root = Puppet.lookup(:root_environment) { nil }
+      root.instance_variable_set(:@static_catalogs, nil) unless root.nil?
```

Comment:
Sorry @joshcooper, I don't remember. It could be there for Puppet Server's functionality to evict caches. There is no use-case in puppet project source base that I know of (unless it is used in tests).